### PR TITLE
Bump Salmon version to 1.9.0

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

File: modules/nf-core/salmon/quant/main.nf [hunk 1]
```diff
@@ -64,6 +65,10 @@ process SALMON_QUANT {
         $args \\
         -o $prefix
 
+    if [ -f $prefix/aux_info/meta_info.json ]; then
+        cp $prefix/aux_info/meta_info.json .
```
Comment (nvnieuwk) [inline modules/nf-core/salmon/quant/main.nf:69]: I may be useful to also incorporate prefix in this filename for distinction purposes?

**Review by nvnieuwk — APPROVED** [review]
A little remark, otherwise LGTM

File: modules/nf-core/salmon/quant/main.nf (hunk 1 above)
Comment (drpatelh) [reply modules/nf-core/salmon/quant/main.nf:69]: Excellent suggestion!

File: modules/nf-core/salmon/quant/main.nf [hunk 2]
```diff
@@ -16,8 +16,9 @@ process SALMON_QUANT {
     val   lib_type
 
     output:
-    tuple val(meta), path("${prefix}"), emit: results
-    path  "versions.yml"              , emit: versions
+    tuple val(meta), path("${prefix}")     , emit: results
+    tuple val(meta), path("meta_info.json"), emit: json_info, optional: true
+    path  "versions.yml"                   , emit: versions
```
Comment (drpatelh) [inline modules/nf-core/salmon/quant/main.nf:21]:
```suggestion
    tuple val(meta), path("${prefix}") , emit: results
    tuple val(meta), path("*info.json"), emit: json_info, optional: true
    path  "versions.yml"               , emit: versions
```

File: modules/nf-core/salmon/quant/main.nf [hunk 3]
```diff
@@ -64,6 +65,10 @@ process SALMON_QUANT {
         $args \\
         -o $prefix
 
+    if [ -f $prefix/aux_info/meta_info.json ]; then
+        cp $prefix/aux_info/meta_info.json .
+    fi
```
Comment (drpatelh) [inline modules/nf-core/salmon/quant/main.nf:70]:
```suggestion
    if [ -f $prefix/aux_info/meta_info.json ]; then
        cp $prefix/aux_info/meta_info.json "${prefix}_meta_info.json"
    fi
```

File: modules/nf-core/salmon/quant/meta.yml [hunk 4]
```diff
@@ -42,10 +42,14 @@ input:
...
 output:
-  - sample_output:
+  - results:
       type: directory
       description: Folder containing the quantification results for a specific sample
       pattern: "${prefix}"
+  - json_info:
+      type: file
+      description: File containing meta information from Salmon quant
+      pattern: "meta_info.json"
```
Comment (drpatelh) [inline modules/nf-core/salmon/quant/meta.yml:52]:
```suggestion
      pattern: "*info.json"
```

**Pull request merged**